plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -142,41 +150,61 @@ private static Type toTrinoType(DataSchema.ColumnDataType columnDataType)
         throw new PinotException(PINOT_UNSUPPORTED_COLUMN_TYPE, Optional.empty(), "Unsupported column data type: " + columnDataType);
     }
 
-    private static List<String> resolvePinotColumns(SchemaTableName schemaTableName, List<String> trinoColumnNames, Map<String, ColumnHandle> columnHandles)
+    private static List<PinotColumnHandle> getPinotColumns(SchemaTableName schemaTableName, List<ExpressionContext> expressions, List<String> aliases, Map<String, ColumnHandle> columnHandles, PinotTypeResolver pinotTypeResolver, Map<String, Type> aggregateTypes)
     {
-        ImmutableList.Builder<String> pinotColumnNamesBuilder = ImmutableList.builder();
-        for (String trinoColumnName : trinoColumnNames) {
-            if (trinoColumnName.equals(WILDCARD)) {
-                pinotColumnNamesBuilder.addAll(columnHandles.values().stream().map(handle -> ((PinotColumnHandle) handle).getColumnName()).collect(toImmutableList()));
+        ImmutableList.Builder<PinotColumnHandle> pinotColumnsBuilder = ImmutableList.builder();
+        for (int index = 0; index < expressions.size(); index++) {
+            ExpressionContext expressionContext = expressions.get(index);
+            Optional<String> alias = getAlias(aliases, index);
+            // Only substitute * with columns for top level SELECT *
```

Comment:
Can you explain why?

Reply:
The only times "*" occurs in pql is in `select *` or `count(*)`. Since `select *` is the only time that "*" needs to be substituted with all columns.

Reply:
The top-level makes it confusing for me. Does it mean that in a query like `SELECT * FROM (SELECT * FROM tbl)` you'd only substitute the outer `*`? Or is the assumption that the query would fail anyway since Pinot doesn't support subqueries so DynamicTableBuilder would never see a subquery passed to it?

Reply:
Yep, pinot does not have subqueries so that query would fail during parsing. Should I remove the "top level" part?

Reply:
Instead add something like

`Only substitute * with columns for top level SELECT *. Since Pinot doesn't support subqueries (yet) we can only have one occurence of SELECT *`.

"top-level" was misleading since the code doesn't seem to enforce that.